CDCgov/prime-reportstream
# Add REST for LA-PHL and Refactor

This PR added REST transport support for LA-PHL and refactored it.  I illuminated the hardcode dependencies to URL endpoints of STLTs.  Now, it uses the information from the transport settings to determine the authentication and send reports to STLTs.

Test Steps:
1. ./gradlew clean package
2. docker-compose up
3. Populate the local vault with credentials for the following STLTs:
    DATAINGESTION--DATATEAM-CDC-NBS,  FLEXION--ETOR-SERVICE-RECEIVER, LA-PHL--ETOR-NBS-ORDERS, OK-PHD--ELR, NY-PHD--ELR, NY-PHD--FULL-ELR-TEST, WA-PHD--REST.
4. Upload settings
    ./prime multiple-settings set --input ./settings/staging/0160-nbs-staging-new-rest-setting.yml
   ./prime multiple-settings set --input ./settings/staging/0161-flexion-staging-new-rest-setting.yml
  ./prime multiple-settings set --input ./settings/staging/0162-la-phl-staging-new-rest-setting.yml
  ./prime multiple-settings set --input ./settings/staging/0163-ok-phd-staging-new-rest-setting.yml
  ./prime multiple-settings set --input ./settings/staging/0164-ny-staging-new-rest-setting.yml
5. Run STLT's check as following:
  curl "http://localhost:7071/api/check?sftpcheck=dataingestion.datateam-cdc-nbs"
  Result:
  ![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/4feef47a-a07e-4cd2-b2ef-9df40639e2fc)
 curl "http://localhost:7071/api/check?sftpcheck=flexion.etor-service-receiver"
 Result:
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/919a7155-e31b-49cd-876b-8b0acf428ad0)
curl "http://localhost:7071/api/check?sftpcheck=la-phl.etor-nbs-orders"
Result:
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/4d4ece82-6512-40e0-a97f-aa8d1559a62e)
curl "http://localhost:7071/api/check?sftpcheck=ok-phd.elr"
Result:
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/aee5c18f-57ff-4a18-831f-3e8138e11dfa)
curl "http://localhost:7071/api/check?sftpcheck=ny-phd.elr"
Result:
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/8437f6dd-3168-46a0-b334-87b818e70d46)
curl "http://localhost:7071/api/check?sftpcheck=ny-phd.full-elr-test"
Result:
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/51660358-fef1-4724-a0f0-dfa8e72396e6)
curl "http://localhost:7071/api/check?sftpcheck=wa-phd.rest"
Result:
![image](https://github.com/CDCgov/prime-reportstream/assets/20068283/cd4b2088-a5c4-45d7-b9bc-991d5edb7298)

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing
- [x] Ran ./gradlew testIntegration

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
